src/expr.rs

```diff
@@ -435,7 +435,7 @@ pub(crate) fn format_expr(
     };
 
     expr_rw
-        .and_then(|expr_str| recover_comment_removed(expr_str, expr.span, context))
+        .and_then(|expr_str| Some(recover_comment_removed(expr_str, expr.span, context)))
```

Comment:
It would be nicer in my opinion to call `map` here so that you don't need to wrap the value in `Some`:
```suggestion
        .map(|expr_str| recover_comment_removed(expr_str, expr.span, context))
```